invenio_records_resources/services/records/service.py

```diff
@@ -552,3 +558,37 @@ def on_relation_update(
 
             self.reindex(identity, search_query=search_query)
         return True
+
+    @unit_of_work()
+    def reindex_latest_first(
+        self, identity, search_preference=None, extra_filter=None, uow=None
+    ):
+        """Reindexes records matching the query filter, prioritizing latest versions.
```

Comment:
I would add as a comment that this should happen in a celery task, and not in sync request, given that `scan` might be slow.